Scripts/Player_Scripts/player.gd

```diff
@@ -13,13 +19,45 @@ func _input(event):
 		
 func _ready():
 	Input.set_mouse_mode(Input.MOUSE_MODE_CAPTURED)
-	
 
 func _physics_process(delta: float) -> void:
 	_apply_gravity(delta)
 	_jump()
 	_handle_movement()
 	move_and_slide()
+	
+	# Sticking flashlight code here for now
+	# Stops charging down once charge is at 0
+	if (light_charge > 0 && flash_light_on == true):
+		light_charge -= delta
+	
+	# Checks if light charge is under the fading threshold, default 40%
+	# then starts fading the flashlight
+	if (flash_light_on == true):
```

Comment:
flash_light_on is already a boolean so no need for a comparison

if flash_light_on:
  #do things